orchestrator/execout/execout_walker.go

```diff
@@ -115,10 +105,14 @@ func computeNewWait(previousWait time.Duration, storeIsLocal bool) time.Duration
 	return newWait
 }
 
-func (r *Walker) sendItems(sortedItems []*pboutput.Item) error {
-	for _, item := range sortedItems {
+func (r *Walker) sendItems(get func() (*pboutput.Item, error)) error {
+	for {
+		item, err := get()
+		if err != nil {
+			return err
+		}
 		if item == nil {
-			continue // why would that happen?!
+			return nil // why would that happen?!
 		}
```

Comment:
If the contract is for get to return either `<non-nil>, nil` or `nil, <error>`, we should panic here. 

Also, you should look at new Go iterator interface, we could implement it and consuming for greater conformance to how iterator should be defined now in Go.